### PR TITLE
[Mobile Payments] Add experimental flag for In-Person Payments in Canada

### LINKED ISSUE #6034: [Mobile Payments] Extend feature flag to use experimental feature toggles
As suggested in https://github.com/woocommerce/woocommerce-ios/pull/5992#pullrequestreview-864765217, add a new experimental features toggle to be able to test dev/alpha builds without IPP in Canada

### PR DESCRIPTION
**Pull request by koke**
Closes: #6034 

### Description

Adds a new Experimental Feature flag for In-Person Payments in Canada.

The PR got a little large because all the boilerplate required to manage App Settings, but it's mostly copied from the Stripe Extension case. The only thing a bit different is the initialization of `CardPresentPaymentsOnboardingUseCase` now needs to check for two features, so I had to refactor that a little bit.

### Testing instructions

By default, the feature should be off, so switch to a store with WCPay set up in Canada:

1. Go to Settings -> In-Person Payments, it should say country not supported
1. Go to Settings -> Experimental Features, turn on In-Person Payments in Canada
1. Go to Settings -> In-Person Payments, it should show options to order/manage readers
1. Go to Settings -> Experimental Features, turn off In-Person Payments in Canada
1. Go to Settings -> In-Person Payments, it should say country not supported

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift:71]: 👍 handy

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift:383]: Github's highlighting here makes me think the comment is about the configuration going to heaven 😅

File: WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift
Comment (joshheald) [inline WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: These tests do seem a bit brittle and annoying don't they... Perhaps we can do this instead:
```suggestion
        let action = stores.receivedActions.first(where: { $0 is CardPresentPaymentAction }) as? CardPresentPaymentAction
```

I've tried this locally and it works

**Review by joshheald — APPROVED** [review]
Nice, thanks for doing this!  Tests well.

One specific test suggestion in the comments.

Only other thing - did you consider removing the feature flag and instead checking for `.localDeveloper || .alpha` where we show the experimental toggle? It could help to avoid confusion/accidental use of the Feature flag rather than the toggle to gate features. I don't mind but just a thought 😊

File: WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift
Comment (koke) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: I guess it depends on what we want to test exactly and both options make some assumptions. I think I prefer the existing one because the last action should be `use`. If we wanted to allow for more flexibility, we'd still want to get the last `use` not the first. If I hadn't done the optimization where `updateState` is only called once on initialization, we'd be getting two `use` actions if the store had both features enabled.

Comment (koke) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: > Only other thing - did you consider removing the feature flag and instead checking for `.localDeveloper || .alpha` where we show the experimental toggle? It could help to avoid confusion/accidental use of the Feature flag rather than the toggle to gate features. I don't mind but just a thought 😊

I think that's a valid concern, but I'm not sure I want to start using `BuildConfiguration` values outside the feature flag service. Maybe the ideal solution would be a system that takes the two concepts (feature flag and experimental feature toggles), and makes them work more cohesively. Though it feels hard to justify investing time on it, since it's code that will end up being deleted in the short term.

Comment (joshheald) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: > the last action should be `use`. If we wanted to allow for more flexibility, we'd still want to get the last `use` not the first. If I hadn't done the optimization where `updateState` is only called once on initialization, we'd be getting two `use` actions if the store had both features enabled.

Makes sense... perhaps just worth removing the assert on the count of actions then?

Comment (joshheald) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: > I'm not sure I want to start using `BuildConfiguration` values outside the feature flag service. 

Fair enough... instead of larger rework, perhaps we should add a shared feature flag called `shouldShowAlphaExperimentalToggles`, which is set to `.localDeveloper || .alpha`, and can be used for multiple toggles in the experimental features screen?

Comment (koke) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: That would make a lot of sense I think. I think a long time ago `BuildConfiguration` supported beta and internal builds, so there were a few more possibilities, but these days we seem to always toggle between dev+alpha only and including production, so maybe we can use a simpler system 🤔 

Out of scope for this PR, but I'd be up for a proposal like that

Comment (koke) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: > Makes sense... perhaps just worth removing the assert on the count of actions then?

Sorry, I saw this last comment after hitting the merge button, and also realized I nested my comments in the wrong place 🤦🏽

Comment (koke) [reply WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsOnboardingUseCaseTests.swift:383]: Removed those checks in #6046

**Pull request merged**
